app/src/org/commcare/android/view/EntityView.java

```diff
@@ -530,5 +491,8 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
                 views[i].setLayoutParams(params);
             }
         }
-    }*/
+        
+        // Use children's new widths
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

```

Comment:
I'm curious, do you think it would work to replace the second super call with setMeasuredDimension?